### PR TITLE
website/integrations: vault: add external group documentation

### PR DESCRIPTION
**Pull request by mzhaase**
## Details

This PR changes the documentation page https://docs.goauthentik.io/integrations/services/hashicorp-vault/.
It adds more in-depth explanation on how to integrate authentik with hashicorp vault when wanting to use external groups.

---

### CODE REVIEW

File: website/integrations/services/hashicorp-vault/index.md [hunk 1]
```diff
@@ -88,8 +88,64 @@ vault write auth/oidc/role/reader \
       policies="reader"
 ```
 
+## External Groups
+
+If you wish to manage group membership in vault via Authentik you have to use [external groups](https://developer.hashicorp.com/vault/tutorials/auth-methods/oidc-auth#create-an-external-vault-group).
```
Comment (tanberry) [inline website/integrations/services/hashicorp-vault/index.md:93]:
```suggestion
If you wish to manage group membership in Hashicorp Vault via Authentik you have to use [external groups](https://developer.hashicorp.com/vault/tutorials/auth-methods/oidc-auth#create-an-external-vault-group).
```

File: website/integrations/services/hashicorp-vault/index.md [hunk 2]
```diff
@@ -88,8 +88,64 @@ vault write auth/oidc/role/reader \
...
+
 :::note
-If you intend to create [external groups](https://developer.hashicorp.com/vault/tutorials/auth-methods/oidc-auth#create-an-external-vault-group) in Vault to manage user access the OIDC role will need to specifically request a custom scope using the `oidc_scopes` option when creating the OIDC role.
```
Comment (tanberry) [inline website/integrations/services/hashicorp-vault/index.md:92]: Does one no longer need to use the `oidc_scopes` option at all?

File: website/integrations/services/hashicorp-vault/index.md [hunk 3]
```diff
@@ -88,8 +88,64 @@ vault write auth/oidc/role/reader \
...
+This assumes that the steps above have already been completed and tested.
 :::
+
+### Step 1
+
+In authentik, edit the oidc provider created above. Unser "Advanced protocol settings" add "authentik default OAuth Mapping: OpenID 'profile'". This includes the "groups" mapping.
+
+### Step 2
+
+In hashicorp vault, change the reader role
```
Comment (tanberry) [inline website/integrations/services/hashicorp-vault/index.md:105]:
```suggestion
In Vault, change the reader role to have the following settings:
```

File: website/integrations/services/hashicorp-vault/index.md [hunk 4]
```diff
@@ -88,8 +88,64 @@ vault write auth/oidc/role/reader \
...
+      allowed_redirect_uris="https://vault.company/ui/vault/auth/oidc/oidc/callback" \
+      allowed_redirect_uris="https://vault.company/oidc/callback" \
+      allowed_redirect_uris="http://localhost:8250/oidc/callback" \
+      user_claim="sub" \
+      policies="reader" \
+      groups_claim="groups" \
+      oidc_scopes=[ "openid profile email" ]
+```
+
+Add a group
```
Comment (tanberry) [inline website/integrations/services/hashicorp-vault/index.md:119]:
```suggestion
Add a group.
```

File: website/integrations/services/hashicorp-vault/index.md [hunk 5]
```diff
@@ -88,8 +88,64 @@ vault write auth/oidc/role/reader \
...
+
+```
+vault write identity/group/reader \
+    name="reader" \
+    policies=["reader"] \
+    type="external"
+```
+
+Get the canonical id of the group
```
Comment (tanberry) [inline website/integrations/services/hashicorp-vault/index.md:128]:
```suggestion
Get the canonical id of the group.
```

Comment (tanberry) [reply website/integrations/services/hashicorp-vault/index.md:128]: If this is the canoncal ID, let's add this full name: `vault_identity_group`

File: website/integrations/services/hashicorp-vault/index.md [hunk 6]
```diff
@@ -88,8 +88,64 @@ vault write auth/oidc/role/reader \
...
+
+```
+vault list identity/group/id
+```
+
+Get the id of the oidc accessor
```
Comment (tanberry) [inline website/integrations/services/hashicorp-vault/index.md:134]:
```suggestion
Get the ID of the OIDC accessor.
```

File: website/integrations/services/hashicorp-vault/index.md [hunk 7]
```diff
@@ -88,8 +88,64 @@ vault write auth/oidc/role/reader \
...
+
+```
+vault auth list
+```
+
+
+Add a group alias, this maps the group to the oidc backend
```
Comment (tanberry) [inline website/integrations/services/hashicorp-vault/index.md:141]:
```suggestion
Add a group alias; this alias maps the group to the OIDC backend.
```

File: website/integrations/services/hashicorp-vault/index.md [hunk 8]
```diff
@@ -88,8 +88,64 @@ vault write auth/oidc/role/reader \
...
+
+```
+vault write identity/group-alias \
+    mount_accessor="auth_oidc_xxxxxx" \
+    canonical_id="group_id" \
+    name="group name in authentik"
+```
+
 You should then be able to sign in via OIDC
```
Comment (tanberry) [inline website/integrations/services/hashicorp-vault/index.md:150]: Can you add a period at end of this sentence, please?

**Review by tanberry — CHANGES_REQUESTED** [review]
Thanks so much for adding this additional, important info! A couple of edits and suggestions, then let's git it merged! Thanks again.

**Comment by tanberry** [comment]
You might need to run `npm prettier` or `make website` and then push again, @mzhaase ... the linter is failing on the build. ;-(

**Comment by mzhaase** [comment]
I implemented the changes in a new commit.

File: website/integrations/services/hashicorp-vault/index.md [hunk 9]
```diff
@@ -88,8 +88,63 @@ vault write auth/oidc/role/reader \
...
+Add a group alias, this maps the group to the OIDC backend.
+
+```
+vault write identity/group-alias \
+    mount_accessor="auth_oidc_xxxxxx" \
+    canonical_id="group_id" \
+    name="group name in authentik"
+```
+
+You should then be able to sign in via OIDC.
```
Comment (tanberry) [inline website/integrations/services/hashicorp-vault/index.md:149]: @mzhaase There's some weird formatting going on here.. if you look at the Deploy Preview, line 150 is mushed up on same line as 149. Do we still need 150? Did it get lost from somewhere else?

Comment (tanberry) [reply website/integrations/services/hashicorp-vault/index.md:149]: Screenshot 2024-12-02 at 9 10 51 PM

**Review by tanberry — APPROVED** [review]

**Pull request merged**